### ISSUE TITLE
send_message API exception error: no try clause matching

### ISSUE THREAD

**Issue by maheshn393** [processone/ejabberd#3555]
**## Environment**

- EJabberd 21.1.0 (running latest docker image)
- Erlang version: erlang is not installed in docker image.
- Ubuntu 20.04.1 LTS (GNU/Linux 5.4.0-1029-aws x86_64)
- Installed from: dockerhub (ejabberd:ecs:latest)

**## Configuration** 

- ejabberd.yml per latest docker image.

**## Errors from error.log/crash.log**

Please note this error appears on the console and not in the error.log.

```
~ $ ./bin/ejabberdctl send_message chat joe@localhost jenny@localhost test 'This is a test message'  
Unhandled exception occurred executing the command:
** exception error: no try clause matching 
                  {message,<<"18188271455089416442">>,chat,<<>>,
                           {jid,<<"joe">>,<<"localhost">>,<<>>,<<"joe">>,
                                <<"localhost">>,<<>>},
                           {jid,<<"jenny">>,<<"localhost">>,<<>>,<<"jenny">>,
                                <<"localhost">>,<<>>},
                           [{text,<<>>,<<"test">>}],
                           [{text,<<>>,<<"This is a test message">>}],
                           undefined,[],#{}}
   in function  mod_admin_extra:send_message/5 (src/mod_admin_extra.erl, line 1485)
   in call from ejabberd_ctl:call_command/4 (src/ejabberd_ctl.erl, line 315)
   in call from ejabberd_ctl:try_call_command/4 (src/ejabberd_ctl.erl, line 280)
   in call from ejabberd_ctl:process2/4 (src/ejabberd_ctl.erl, line 233)
   in call from ejabberd_ctl:process/2 (src/ejabberd_ctl.erl, line 211)
   in call from rpc:'-handle_call_call/6-fun-0-'/5 (rpc.erl, line 197)
```

**## Bug description**

Steps to reproduce:

1. Run latest docker image, and get into docker container's shell
2. use ejabberdctl to "register" 2 users on localhost
3. use ejabberdctl to "send_message" from user1 to user2.

See below commands in the order executed after I entered the ejabberd docker container shell.

```
~ $ ./bin/ejabberdctl status
The node ejabberd@c089ab4945d4 is started with status: started
ejabberd 21.1.0 is running in that node

~ $ ./bin/ejabberdctl register joe localhost 123
User joe@localhost successfully registered

~ $ ./bin/ejabberdctl register jenny localhost 123
User jenny@localhost successfully registered

~ $ ./bin/ejabberdctl registered_users localhost
jenny
joe

~ $ ./bin/ejabberdctl send_message chat joe@localhost jenny@localhost test 'This is a test message'  
Unhandled exception occurred executing the command:
```

All pointers appreciated that help resolve issue. I can provide additional clarifications or information as you need. 
Thank you.

**Comment by badlop** [comment]
First of all: your detailed reproduction steps are great! :)

Second: It's a bug introduced in ejabberd 21.01. It was fixed in git master past month (https://github.com/processone/ejabberd/commit/b3d9c0d1f7f7037c1a22161c857c931f5e4b7ed3), so the fix will be included in the next release. In the meantime, you can use a previous ejabberd version, or apply that small patch yourself.

**Comment by maheshn393** [comment]
> First of all: your detailed reproduction steps are great! :)
> 
> Second: It's a bug introduced in ejabberd 21.01. It was fixed in git master past month ([b3d9c0d](https://github.com/processone/ejabberd/commit/b3d9c0d1f7f7037c1a22161c857c931f5e4b7ed3)), so the fix will be included in the next release. In the meantime, you can use a previous ejabberd version, or apply that small patch yourself.

Thank you @badlop. Much appreciated.
